### ISSUE TITLE
Temporary files/directory problem with FICO-Xpress in Windows

### ISSUE THREAD

**Issue by PetrouGeorge** [coin-or/pulp#148]
Hi all,

I'm opening a new issue, since I identified the source of the problem mentioned [here](https://github.com/coin-or/pulp/issues/143).

I'm using a Windows 7 Pro 64-bit system where I have installed FICO-Xpress 8.2, Python 3.5 and PuLP 1.6.5. In order to solve the optimisation problem with FICO-Xpress, PuLP has to
1. create a `.lp` file,
2. execute `optimizer.exe` to solve it and
3. read the `.prt` to retrieve the solution.

However this doesn't work in Windows because it seems that PuLP doesn't manage properly the name of the temporary directories and filenames with spaces. At least that was the case for me. So, I had to activate the `keepFiles` option and use a simple name for my problem, since this name is also used to create the `.lp` and `.prt` files.

Another thing that I had to do was to activate the `msg` option. I have no idea why, but when it isn't activated, PuLP doesn't work.

To summarize :
1. the name of the problem has to be simple, e.g. `prob = LpProblem("TheWhiskasProblem",LpMinimize)` and
2. the msg and keepFiles options have to be activated : `prob.setSolver(XPRESS(msg=1, keepFiles=True))`.

I hope that works for you too.

**Comment by stumitchell** [comment]
Great I'll see if I can change the code thanks for looking at this

**Comment by s-c-e** [comment]
Hi @PetrouGeorge - I worked with Xpress for the first time yesterday - I can confirm your experience. Python 3.4 32-bit, Windows, Xpress 8.4.

With ```msg=1``` and SPACES in the model name, the log looked like this:
```
FICO Xpress Solver 64bit v8.4.4 Jan 31 2018
(c) Copyright Fair Isaac Corporation 1983-2017. All rights reserved
 Optimizer v32.01.07    [C:\xpressmp\bin\xprs.dll]
?85 Error: File not found: C:UsersXXXXAppDataLocalTempcdeb7a6a60414716a2ac4d83ce02280c-pulp.lp
?91 Error: No problem has been input
?1055 Error: Can not resume global search - not currently solving a MIP.
?91 Error: No problem has been input
FAILED
```
Removing spaces from the model name resulted in a valid path to the LP file, but it failed with a different message (I thought I saved the log from this, but can't find it now)...

I can confirm that setting ```keepFiles=1``` is a valid work-around for getting things going.
